src/query/storages/fuse/src/operations/changes.rs

```diff
@@ -153,56 +154,64 @@ impl FuseTable {
                 )
             }
             StreamMode::Standard => {
-                let a_table_alias = format!("_change_insert${}", suffix);
-                let a_cols = cols.join(", ");
+                let quote = ctx.get_settings().get_sql_dialect()?.default_ident_quote();
 
+                let a_table_alias = format!("_change_insert${}", suffix);
                 let d_table_alias = format!("_change_delete${}", suffix);
-                let d_cols = cols
-                    .iter()
-                    .map(|s| format!("d_{}", s))
-                    .collect::<Vec<_>>()
-                    .join(", ");
 
+                let mut a_cols_vec = Vec::with_capacity(cols.len());
+                let mut d_alias_vec = Vec::with_capacity(cols.len());
+                let mut d_cols_vec = Vec::with_capacity(cols.len());
+                for col in cols {
+                    a_cols_vec.push(format!("{quote}{col}{quote}"));
+                    d_alias_vec.push(format!("{quote}{col}{quote} as d_{col}"));
+                    d_cols_vec.push(format!("d_{col}"));
+                }
+                let a_cols = a_cols_vec.join(", ");
+                let d_cols_alias = d_alias_vec.join(", ");
+                let d_cols = d_cols_vec.join(", ");
+
+                let cte_name = format!("_change${}", suffix);
                 format!(
-                    "with _change({a_cols}, change$action, change$row_id, \
-                        {d_cols}, d_change$action, d_change$row_id) as \
+                    "with {cte_name} as materialized \
```

Comment:
It is recommended to remove `materialized` because it has not been fully tested.

Reply:
> It is recommended to remove `materialized` because it has not been fully tested.

Hi, @Dousir9 The issue was caused by the removal of `materialized`https://github.com/databendlabs/databend/pull/16900. `FULL OUTER JOIN` wraps fields with `nullable`, leading to type mismatch errors.  
The solution is to reintroduce `materialized` and store `consume stream` information in `QueryContext`, ensuring the atomicity of table metadata updates.  
Additionally, we bypassed the validity check for `WITH consume` by examining specific `m_cte` names.  

As for the previously encountered `panic`(https://github.com/databendlabs/databend/actions/runs/12064856343/job/33643414721?pr=16900), it was caused by a bug in `UNION ALL`, which will be fixed in a separate PR.